### PR TITLE
Updated Immunization DSTU2 Resource

### PR DESCRIPTION
**Pull request by kckeim**
Just a handful of changes on this one, including some re-wording of Overview and Implementation Notes sections to be more descriptive / in a similar style to other resources that were recently updated.
Verified good links through document.
Validated that API calls behave as expected with both the expected and unexpected parameters.

Description
----
- Expanded the Overview Section, and more closely matches the verbiage used by HL7.org itself now.
  - Now includes direct links out to the `AllergyIntolerance` and `MedicationAdministration` resources.
- Consolidated Disclaimer, Error, and Outcome sections towards the top.
- Adjusted parameters table's details.
- Expanded on Implementation Notes to match the performance of the API, and explain requirements and expected behavior of each search parameter.

*****************************
PR Checklist
----
- [X] Screenshot(s) of changes attached before changes merged.
- [X] Screenshot(s) of changes attached after changes merged and published.
*****************************
Old Overview:
image

New Overview with links:
image
*****************************
Old parameters + Implementation Notes:
image

New parameters + Implementation Notes:
image

### CODE REVIEW

**Review by RicardoCrespo-ora — APPROVED** [review]

**Review by summerlcollins — APPROVED** [review]
approved per [PRODOC-202500](https://jira.cerner.com/browse/PRODOC-202500)

File: content/millennium/dstu2/medications/immunization.md [hunk 1]
```diff
@@ -9,11 +9,11 @@ title: Immunization | DSTU 2 API
 
 ## Overview
 
-The Immunization resource includes the view of current and historical administration of vaccinations to a patient in all healthcare settings. This resource contains the functionality to query a patient's immunization history.
+The Immunization resource is intended to cover the recording of current and historical administration of vaccines to patients across all healthcare disciplines, in all care settings, and all regions.  This resource contains the functionality to query a patient's immunization history. Detailed administration records may be found in the [`MedicationAdministration`] resource, while this Immunization resource would represent the known vaccination history regardless of where the administration itself was done.
 
-Detailed administration records may be found in MedicationAdministration, while the Immunization resource would represent the known vaccination history regardless of where the administration itself was done.
+An immunization reaction may indicate an allergy or intolerance. If so, a separate [`AllergyIntolerance`] resource instance must be created.
```
Comment (amcginn) [inline content/millennium/dstu2/medications/immunization.md:14]: this is a weird business rule/recommendation that I don't think I've seen elsewhere.

@cldeitrich, thoughts on how to approach this one?

Comment (amcginn) [reply content/millennium/dstu2/medications/immunization.md:14]: looks like this is replicating information from HL7:
https://hl7.org/fhir/DSTU2/immunization.html#bnr

File: content/millennium/dstu2/medications/immunization.md [hunk 2]
```diff
@@ -49,15 +55,27 @@ Search for Immunizations that meet supplied query parameters:
...
+-----------|---------------|---------------|-----------------------------------------------------------------------------------------------------
+ `_id`     | Conditionally | [`token`]     | The logical resource ID associated with the resource. This parameter is required if the `patient` parameter is not used. Example: `12345`
+ `patient` | Conditionally | [`reference`] | The patient for the vaccination record(s). This parameter is required if the `_id` parameter is not used. Example: `12345`
+ `date`    | No            | [`date`]      | Date range for the immunization administration(s). Example: `date=ge2020-01-01T08:00:00.000Z&date=le2020-01-31T17:00:00.000Z`
+
+_Implementation Notes_
 
-Notes:
+- When searching with the `_id` parameter:
+  - It can be provided with either a single reference, or a comma-separated list of references. Example `_id=1234` or `_id=1234,5678`
```
Comment (amcginn) [inline content/millennium/dstu2/medications/immunization.md:67]: I forget where we had this conversation. do we need to include this on every resource?

**Review by amcginn — APPROVED** [review]

**Comment by Someshnsn** [comment]
HI @kckeim @summerlcollins 
Should we make this changes for r4
https://fhir.cerner.com/millennium/r4/clinical/medications/immunization/

**Review by Someshnsn — CHANGES_REQUESTED** [review]
HI @kckeim @summerlcollins
Should we make this changes even for r4?
https://fhir.cerner.com/millennium/r4/clinical/medications/immunization/

**Comment by Someshnsn** [comment]
HI

> HI @kckeim @summerlcollins Should we make this changes even for r4? https://fhir.cerner.com/millennium/r4/clinical/medications/immunization/

Could see the r4 changes, approving the PR
https://github.com/cerner/fhir.cerner.com/pull/1004

**Review by Someshnsn — APPROVED** [review]

**Pull request merged**